code/modules/client/client procs.dm

```diff
@@ -454,3 +454,56 @@ NOTE:  You will only be polled about this role once per round. To change your ch
 		winset(usr, "mainwindow.mainvsplit", "right=mapwindow;left=rpane;splitter=[newsplit]")
 	else
 		winset(usr, "mainwindow.mainvsplit", "right=rpane;left=mapwindow;splitter=[newsplit]")
+
+/client/proc/update_special_views()
+	var/list/vi = view(world.view,get_holder_at_turf_level(mob))
+
+	if(prefs.space_parallax)	//Updating parallax for clients that have parallax turned on.
+		for(var/turf/space/T in vi)
+			if(!parallax_initialized)
+				break
+			else
+				mob.hud_used.update_parallax_values()
+				break
+
+	for(var/obj/structure/window/W in vi)
+		if(W.one_way)
+			update_one_way_windows(vi)	//Updating the one-way window overlay if the client has one in view.
+			break
+
+/client/proc/update_one_way_windows(var/list/v)		//Needed for one-way windows to work.
+	var/Image										//Code heavily cannibalized from a demo made by Byond member Shadowdarke.
+	var/turf/Oneway
+	var/obj/structure/window/W
+	var/list/newimages = list()
+	var/list/onewaylist = list()
+	var/inverse_dir
+
+	if(!v)
+		return
+
+	ObscuredTurfs.len = 0
+
+	for(W in v)
+		if(W.one_way)
+			inverse_dir = turn(W.dir, 180)
+			if(inverse_dir & get_dir(W,get_holder_at_turf_level(mob)))
+				Oneway = get_turf(W)
+				Oneway.opacity = 1
+				onewaylist += Oneway
+
+	if(onewaylist.len)
+		var/list/List = v - view(world.view,get_holder_at_turf_level(mob))
```

Comment:
The variable must be assigned because I need to compare the tiles visible before making all the tiles behind the windows opaque with the tiles visible after making all the windows opaque, in order to get a list of the tiles that *would* be obscured if the windows were opaque, so that I can manually obscure them afterward.